lib/rust/ensogl/core/src/display/symbol/gpu.rs

```diff
@@ -494,6 +497,44 @@ impl From<&Symbol> for SymbolId {
 }
 
 
+// === Weak References ===
+
+/// Weak reference to a [`Symbol`].
+#[derive(Debug, Clone)]
+pub struct WeakSymbol {
+    data:         Weak<SymbolData>,
+    shader_dirty: WeakShaderDirty,
+    shader:       WeakShader,
+}
+
+impl WeakElement for WeakSymbol {
+    type Strong = Symbol;
+
+    fn new(view: &Self::Strong) -> Self {
+        let data = view.data.downgrade();
+        let shader_dirty = view.shader_dirty.downgrade();
+        let shader = view.shader.downgrade();
+        Self { data, shader_dirty, shader }
+    }
+
+    fn view(&self) -> Option<Self::Strong> {
+        let data = self.data.upgrade()?;
+        let shader_dirty = self.shader_dirty.upgrade()?;
+        let shader = self.shader.upgrade()?;
+        Some(Symbol { data, shader_dirty, shader })
+    }
+
+    fn is_expired(&self) -> bool {
+        self.data.is_expired()
+    }
+
```

Comment:
The dirty flag owns a callback; if that callback strongly-referenced a `WeakSymbol`, it could cause a leak. The way these types are currently used that will not happen, but I think if a type is "weak", developers should be able to rely on it never participating in reference cycles.

Reply:
Ok, could you add docs to explain the design in the code then please? Because without that its not clear why we have this weak ref there as its not preventing any cycles.